### PR TITLE
GSoC 2015 : JuliaQuantum : Framework for Solvers

### PR DESCRIPTION
**Pull request by amitjamadagni**
Hello everyone,
       The project aims to equip JuliaQuantum with a framework to solve dynamical equations arising in quantum mechanics. Given an opportunity, I would be working under the mentroship of @acroy. Further details have been linked in the proposal.

### CODE REVIEW

File: 2015/proposals/Amit.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+My journey with open source started with SymPy where I had fixed some bugs in Quantum physics module, my interest in Topological Quantum Computation introduced me to the low dimensional topological based ideas of Links/Knots which further led to a Google Summer of Code 2014 project for Sage Mathematical Software where I was involved in developing Computational Knot theory library. The project was based on python and it mainly dealt with conversions between various representations of Links/Knots, on the top of it there were few invariants implemented to distinguish between various Links/Knots. In addition my interest towards hardware allowed me undertake projects involving Raspberry Pi, the details of which could be found on my GitHub repo (Repo titles : [PiSMS : sending SMS using Pi] (https://github.com/amitjamadagni/PiSMS), [Project : Controlling GPIO pins via wireless communication techniques] (https://github.com/amitjamadagni/Project)). I had an opportunity to contribute to Julia and Julia Quantum in the form of enhancing few features. This project would further enhance my knowledge in various domains ranging from Quantum Mechanics to further understanding of Julia.
```
Comment (r-gaia-cs) [inline 2015/proposals/Amit.md]: "The project was based on python" -> "The project was based on Python"

File: 2015/proposals/Amit.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+* GitHub Profile : https://github.com/amitjamadagni
+* GSoC 2014 : https://www.google-melange.com/gsoc/project/details/google/gsoc2014/amitjamadagni/5676830073815040
+* GSoC 2014 blog : https://knotsknotted.wordpress.com/category/gsoc/
+* Contribution to Julia : Implementation of ror! and rol! (https://github.com/JuliaLang/julia/pull/9822)         
+* Contribution to Julia Quantum : Normalize to QuArray (https://github.com/JuliaQuantum/QuBase.jl/pull/15)           
+
+---
+### Project Synopsis : 
+
+The aim of the project is to equip the [JuliaQuantum] (http://juliaquantum.github.io/) libraries with solvers which are on par with QuTiP. To this end, a flexible and unified framework will be created, which allows to access different solvers. The project will make use of existing packages (if possible) and aims to contribute back in order to homogenize their interfaces and improve the performance.
```
Comment (r-gaia-cs) [inline 2015/proposals/Amit.md]: > The project will make use of existing packages (if possible)

Do you have any package in mind?

File: 2015/proposals/Amit.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+Main goals of the project include:
+
+1. Framework to solve Schrodinger (SE) and Quantum Master Equations (ME) by
+    * Exponential solvers
+    * ODE solvers (Runge-Kutta(RK) Methods)
+    * Monte Carlo Wave Function Method (MCWF)
+    * Steady State Methods
+
+2. Parallelizing some of the above tasks (Integrating QuArrays and methods involved in MCWF) using the existing infrastructure for parallel computing features in Julia to achieve greater efficiency.
```
Comment (r-gaia-cs) [inline 2015/proposals/Amit.md]: Are you going to use one library to solve the equations or will you wrote your own solver? If you are going to use one library I'm not sure if you can take advantage from Julia parallel computing features.

File: 2015/proposals/Amit.md (hunk 2 above)
Comment (amitjamadagni) [inline 2015/proposals/Amit.md]: @r-gaia-cs I have linked to the packages in the Implementation section under Details. Please do let me know if I need to include it here as well. Thanks.

**Comment by r-gaia-cs** [comment]
@amitjamadagni Thanks for your proposal.

Great that you already submit your proposal to https://www.google-melange.com/gsoc/homepage/google/gsoc2015. You can change it until March 27th 19:00 UTC.

File: 2015/proposals/Amit.md (hunk 3 above)
Comment (amitjamadagni) [inline 2015/proposals/Amit.md]: @r-gaia-cs the framework for the solvers would use various libraries for solvers (though not necessary for all solvers) as well as for parallel computing features. As we have the construct of `QuArray` based on arrays we could parallelize at a very basic level, as well the parallelization can be achieved in implementing the quantum monte carlo wave function methods which is at a higher level of abstraction. Hope I was able to convey the ideas.

**Comment by amitjamadagni** [comment]
@r-gaia-cs Thank you for the review and information. Please do let me know if anything more could be done from my side. Thanks once again.

**Comment by r-gaia-cs** [comment]
Thanks for the new information. My suggest is that you update it at https://www.google-melange.com/gsoc/homepage/google/gsoc2015.

**Comment by r-gaia-cs** [comment]
I'm merging this issue since student application period is over.

**Pull request merged**